pandas/core/interval.py

```diff
@@ -0,0 +1,413 @@
+            raise ValueError('left and right must have the same length')
+        if not (self.left <= self.right).all():
+            raise ValueError('left side of interval must be <= right side')
+
+    def _simple_new(cls, values, name=None, **kwargs):
+        # ensure we don't end up here (this is a superclass method)
+        raise NotImplementedError
+
+    def _cleanup(self):
+        pass
+
+    @property
+    def _engine(self):
+        raise NotImplementedError
+
+    @cache_readonly
+    def _tree(self):
+        return IntervalTree(self.left, self.right, closed=self.closed)
+
+    @property
+    def _constructor(self):
+        return type(self).from_intervals
+
+    @classmethod
+    def from_breaks(cls, breaks, closed='right', name=None):
+        return cls(breaks[:-1], breaks[1:], closed, name)
+
+    @classmethod
+    def from_intervals(cls, data, name=None):
+        data = np.asarray(data)
+        left, right, closed = intervals_to_interval_bounds(data)
+        return cls(left, right, closed, name)
+
+    @classmethod
+    def from_tuples(cls, data, closed='right', name=None):
+        left = []
+        right = []
+        for l, r in data:
+            left.append(l)
+            right.append(r)
+        return cls(np.array(left), np.array(right), closed, name)
+
+    def to_tuples(self):
+        return Index(com._asarray_tuplesafe(zip(self.left, self.right)))
+
+    @property
+    def left(self):
+        return self._left
+
+    @property
+    def right(self):
+        return self._right
+
+    @property
+    def closed(self):
+        return self._closed
+
+    def __len__(self):
+        return len(self.left)
+
+    @cache_readonly
+    def values(self):
+        left = np.asarray(self.left)
+        right = np.asarray(self.right)
+        return interval_bounds_to_intervals(left, right, self.closed)
+
+    def __array__(self, result=None):
+        """ the array interface, return my values """
+        return self.values
+
+    def __array_wrap__(self, result, context=None):
+        # we don't want the superclass implementation
+        return result
+
+    def _array_values(self):
+        return self.values
+
+    def __reduce__(self):
+        return self.__class__, (self.left, self.right, self.closed, self.name)
+
+    def _shallow_copy(self, values=None, name=None):
+        name = name if name is not None else self.name
+        if values is not None:
+            return type(self).from_intervals(values, name=name)
+        else:
+            return self.copy(name=name)
+
+    def copy(self, deep=False, name=None):
+        left = self.left.copy(deep=True) if deep else self.left
+        right = self.right.copy(deep=True) if deep else self.right
+        name = name if name is not None else self.name
+        return type(self)(left, right, closed=self.closed, name=name,
+                          fastpath=True)
+
+    @cache_readonly
+    def dtype(self):
+        return np.dtype('O')
```

Comment:
you need an ``Index`` one (that just returns true for an ``IntervalIndex``, false otherwise (e.g. see what we did for ``CategoricalIndex``). its an ``Index`` introspection method.